sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ValueStateImpl.scala

```diff
@@ -54,18 +56,15 @@ class ValueStateImpl[S](
       throw StateStoreErrors.implicitKeyNotFound(stateName)
     }
 
-    val toRow = keyExprEnc.createSerializer()
-    val keyByteArr = toRow
+    val keyByteArr = keyExprEnc.createSerializer()
```

Comment:
nit: sorry for missing - this seems to be also something we can move out

Reply:
Done